### PR TITLE
Export complete data

### PR DESCRIPTION
**Pull request by dhimmel**
@gwaygenomics has previously brought up the need to export our processed datasets for all observations. https://github.com/cognoma/cancer-data/pull/20#issuecomment-242408331:

> As a general comment, while I think it is definitely good for the ML group to have a single dataset that everyone is working on, restricting it like this may not be the optimal solution. Eventually the data will need to be more fluid and subset on the fly depending on different rules which we will need to define later. (e.g. Unsupervised feature construction should not remove gene expression samples that don't have mutation status)

The motivation of this pull request is that Cognoma is producing the most user-friendly data. We should export the complete datasets to enable many applications. Currently, I'm not planning to upload this data to figshare, but we could (especially once we continuously integrate), but users will be able to generate it for themselves.

### CODE REVIEW

File: scripts/2.TCGA-process.py [hunk 1]
```diff
@@ -232,37 +244,44 @@
...
 
 # Peak at the data matrix
 expr_df.iloc[:5, :5]
 
 
+# In[23]:
+
+# Save complete expression matrix
+path = os.path.join('data', 'complete', 'expression-matrix.tsv.bz2')
+expr_df.to_csv(path, sep='\t', float_format='%.3g', compression='bz2')
```
Comment (gwaybio) [inline scripts/2.TCGA-process.py:263]: Do you know how much the `float_format` changes things here? Does it at all? I have slight concerns that someone using the unformated Xena data may get different results than if they were to use our formatted data

**Review by gwaybio — COMMENTED** [review]
One question, but other than that LGTM

File: scripts/2.TCGA-process.py (hunk 1 above)
Comment (dhimmel) [reply scripts/2.TCGA-process.py:263]: Here are a bunch of consecutive values I pulled out of `data/complete/expression-matrix.tsv.bz2`: 2.72, 0, 0, 0, 3.4, 5.18, 6.94, 6.08, 3.67, 9.61, 8.03, 0, 6.52, 0, 1.7, 8.24, 7.53, 1.08. So you you're three digits of precision, which I definitely wouldn't want to go less than. With `float_format='%.3g'` this file is 238 MB. So yeah, I'd love more precision, large files slow everything down and make data harder to use.

I don't really have any tests to see how much `float_format='%.3g'` affects results compared to `float_format='%.4g'`. @gwaygenomics what do you think?

Comment (gwaybio) [reply scripts/2.TCGA-process.py:263]: `float_format='%.3g'` is _probably_ ok? I was just curious as to your reasoning.

I'd say about half of the raw data entries that are not 0 have 3 decimal points, the other half has 4 (while some have many, which are mostly zeros).

Comment (dhimmel) [reply scripts/2.TCGA-process.py:263]: Yeah I think it's fine, here's the [doc for general format](https://docs.python.org/3.5/library/string.html#formatspec):

> General format. For a given precision `p >= 1`, this rounds the number to p significant digits and then formats the result in either fixed-point format or in scientific notation, depending on its magnitude.

Based on my understanding, '%.3g' will corrupt a value by at most 0.1%, which seems trivial especially for noisy gene expression.

**Review by gwaybio — APPROVED** [review]

**Pull request merged**